### DISCUSSION TITLE
Need help: Failure to create pdf with reference by MikTex

### DISCUSSION THREAD

**Discussion by Ye-Chao-Liu** [James-Yu/LaTeX-Workshop#3137] [Q&A]
I.m a greenhand for VScode. I use MikTex + WinEdt before, so I still use MikTex for VScode.

Actually, I can create `pdf` successfully only by the tool `pdflatex`.
`xelatex` and `texify` (recommended for MikTex in Wiki) always show "Recipe terminated with error."
The output is similar for both of them
> Sorry, but texify did not succeed.
> The log file hopefully contains the information to get MiKTeX going again:
> C:\Users\admin\AppData\Local\MiKTeX\2.9\miktex\log\texify.log

When I want to put the reference with a `bib` file, it also fails.
I first run `pdflatex`, then `bibtex` give a yellow warning
> Recipe terminated with error. Retry building the project.

with output
> I couldn't open file name `XXX.aux'

after that running `pdflatex` again is totally useless.

The only way I can success now is running the recipe `pdf-bib-pdf-pdf`, and after it shows `Recipe terminated with error.` at second step, running `pdflatex` twice.

I don't know what goes wrong, how can finish the recipe `pdf-bib-pdf-pdf` completely and the correct `pdf` file with reference?

**Comment by tamuratak** [comment]
This issue has nothing with LaTeX Workshop.

**Comment by Ye-Chao-Liu** [comment]
so there is some problem with VS code or MikTex?

**Comment by tecosaur** [comment]
Try debugging the LaTeX compiling command instead. You can find out what command is run by the LaTeX workshop settings or log output.

**Comment by Ye-Chao-Liu** [comment]
Thank you. I will try it.
